Source/JavaScriptCore/wasm/js/WebAssemblyModuleRecord.cpp

```diff
@@ -739,9 +754,19 @@ JSValue WebAssemblyModuleRecord::evaluate(JSGlobalObject* globalObject)
             ASSERT(!!m_instance->table(*element.tableIndexIfActive));
 
             const auto& offset = *element.offsetIfActive;
-            const uint32_t elementIndex = offset.isGlobalImport()
-                ? static_cast<uint32_t>(m_instance->instance().loadI32Global(offset.globalImportIndex()))
-                : offset.constValue();
+            uint32_t elementIndex = 0;
+            if (offset.isGlobalImport())
+                elementIndex = static_cast<uint32_t>(m_instance->instance().loadI32Global(offset.globalImportIndex()));
+            else if (offset.isConst())
+                elementIndex = offset.constValue();
+            else {
+                uint64_t result;
+                if (!evaluateConstantExpression(moduleInformation.constantExpressions[offset.constantExpressionIndex()], moduleInformation, Wasm::Types::I32, result)) {
```

Comment:
This code evaluates the const expr for element segment offsets. Note that it doesn't evaluate the actual element items (the reference value stored in the segment) yet. That requires more changes to element segment representations (right now it assumes a vector of function indices... this would need to change to a vector of function index or constant expression index) and there's a separate bug filed for it.

Reply:
Can we get a TODO?

Reply:
Sure, I've added a FIXME comment in `parseElementSegmentVectorOfElements` (in the section parser rather than here, because that's where the biggest changes will be).